cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/services/http/handlers/ProgramLifecycleHttpHandlerTest.java

```diff
@@ -1096,6 +1096,81 @@ public void testSchedules() throws Exception {
+                                                               defaultAppId.getApplication(),
+                                                               defaultAppId.getVersion());
+    Assert.assertEquals(2, actualSchSpecs.size());
+
+    // with workflow, without schedule
+    config = new AppWithSchedule.AppConfig(true, false, false);
+    request = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config, null, null, false);
+    Assert.assertEquals(200, deploy(defaultAppId, request).getStatusLine().getStatusCode());
+
+    // schedule should not be updated
+    actualSchSpecs = listSchedules(TEST_NAMESPACE_META2.getNamespaceId().getNamespace(),
+                                   defaultAppId.getApplication(),
+                                   defaultAppId.getVersion());
+    Assert.assertEquals(2, actualSchSpecs.size());
+
+    // without workflow and schedule, schedule should be deleted
+    config = new AppWithSchedule.AppConfig(false, false, false);
+    request = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config, null, null, false);
+    Assert.assertEquals(200, deploy(defaultAppId, request).getStatusLine().getStatusCode());
+
+    actualSchSpecs = listSchedules(TEST_NAMESPACE_META2.getNamespaceId().getNamespace(),
+                                   defaultAppId.getApplication(),
+                                   defaultAppId.getVersion());
+    Assert.assertEquals(0, actualSchSpecs.size());
+
+    // with workflow and  one schedule, schedule should be added
+    config = new AppWithSchedule.AppConfig(true, true, false);
+    request = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config, null, null, true);
+    Assert.assertEquals(200, deploy(defaultAppId, request).getStatusLine().getStatusCode());
+
+    actualSchSpecs = listSchedules(TEST_NAMESPACE_META2.getNamespaceId().getNamespace(),
+                                   defaultAppId.getApplication(),
+                                   defaultAppId.getVersion());
+    Assert.assertEquals(1, actualSchSpecs.size());
+
+    // with workflow and two schedules, but update-schedules is false, so 2nd schedule should not get added
+    config = new AppWithSchedule.AppConfig(true, true, true);
+    request = new AppRequest<>(
+      new ArtifactSummary(artifactId.getName(), artifactId.getVersion().getVersion()), config, null, null, false);
+    Assert.assertEquals(200, deploy(defaultAppId, request).getStatusLine().getStatusCode());
+
+    actualSchSpecs = listSchedules(TEST_NAMESPACE_META2.getNamespaceId().getNamespace(),
+                                   defaultAppId.getApplication(),
+                                   defaultAppId.getVersion());
+    Assert.assertEquals(1, actualSchSpecs.size());
```

Comment:
For this case it should be better to also assert that that schedule is the same as from earlier. Just assert for the name too.